frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -371,10 +406,22 @@ protected void onStop() {
         Util.log("lifecycle: onStop");
     }
 
+    @Override
+    protected void onRestart() {
+        // This is here so that if the GoService gets killed while the app is in background it
+        // will be started again.
+        if (goService == null) {
+            Intent intent = new Intent(this, GoService.class);
+            bindService(intent, connection, Context.BIND_AUTO_CREATE);
+        }
+        super.onRestart();
+    }
+
     @Override
     protected void onDestroy() {
-        super.onDestroy();
         Util.log("lifecycle: onDestroy");
+        goService.stopServer();
```

Comment:
is onDestroy called only when the app shuts down or can it be called at any time? 

asking because i am not sure shutdown followed by serve actually works. does that scenario even happen?

Reply:
It should be called only when the app shuts down. I added this only to stop the server before it is destroyed. I made some tests and when the app is destroyed there is no service hanging, so should be fine, I guess.

Reply:
I noticed that in case of bad connection the backend shutdown could hang on indefinitely, blocking the service destruction and giving issues when the app is started again. To avoid the issue I just removed that, since everything seems to work fine anyway.